packages/components/src/components/molecules/CardPile/CardPile.tsx

```diff
@@ -1,122 +1,161 @@
+  const classes = useStyles(styleProps);
 
-  const thisCard: CardProps | undefined = pileCards[0];
-  const displayedCard = useMemo(() => {
-    const TagName = component;
-    const cardProps: CardProps = {
-      ...thisCard,
-      disableNativeEvents: true,
-    };
-    return !thisCard ? undefined : (
-      <div data-testid="CardPile_card" key={uuid()} className={classnames(classes.card, classes.firstItem)}>
-        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
-        <TagName {...cardProps} ref={firstCardRef} />
-      </div>
-    );
-  }, [thisCard, classes, component]);
-
-  const backFaceComponent = useMemo(() => {
-    const BackFaceTagName = backFace;
-    return pileCards.length <= 1 ? undefined : (
-      <div data-testid="CardPile-backface" className={classnames(classes.card, classes.lastItem)}>
-        <BackFaceTagName />
-      </div>
-    );
-  }, [pileCards.length, classes, backFace]);
+  const [isSelected, setSelect] = useState(false);
+  const [firstCard, secondCard, ...otherCards] = cards;
 
   const toggleSelect = () => {
     setSelect(!isSelected);
   };
-  const onPileClick = () => toggleSelect();
-  const onPileKeyDown = (event: KeyboardEvent<HTMLDivElement>) => event.keyCode === 32 && toggleSelect();
-  const onTurnFirstCard = () => firstCardRef.current?.flip();
-  const onShuffle = () => {
-    firstCardRef.current?.flip(false);
-    const newPile = shuffleArray(pileCards);
-    setPileCards(newPile);
+
+  const onTurnFirstCard = () => {
+    onCardFlipped(!isFaceUp);
+    setSelect(false);
+  };
+
+  const onShufflePile = () => {
+    onShuffle();
+    setSelect(false);
   };
 
   return (
-    <div
+    <Draggable
+      className={classes.cardPile}
+      position={position}
+      height={height}
+      width={width}
+      boundaries={tableBoundaries}
+      onDragged={onPositionChanged}
+      onClick={toggleSelect}
+      // eslint-disable-next-line react/destructuring-assignment
       data-testid="CardPile"
-      role="button"
-      tabIndex={0}
-      onClick={onPileClick}
-      onKeyDown={onPileKeyDown}
-      className={classnames(classes.cardPile, { [classes.multiItem]: pileCards.length > 1 })}
     >
       {isSelected && (
         <>
           <Overlay />
           <div className={classes.cardPileMenu}>
-            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShuffle} />
+            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShufflePile} />
           </div>
         </>
       )}
-      {displayedCard}
-      {backFaceComponent}
-    </div>
+      {firstCard && (
+        <BaseCard
+          data-testid="CardPile-firstCard"
+          className={classes.firstItem}
+          width={width}
+          height={height}
+          frontFace={frontFace}
+          backFace={backFace}
+          disableNativeEvents
+          faceUp={isFaceUp}
+        />
+      )}
+      {secondCard && (
+        <BaseCard
```

Comment:
Well actually after defining the `frontFace` and `backFace` props for this component, I realised for the second card I cannot just use the `backFace` as it is given in the props, since I need to add some style to it (`className={classes.secondItem}`(. A solution would be to add a third prop like `otherCardBackface` or whatever, and I find it uglier than having a second BaseCard. Also, you can see that the second BaseCard does not have a front face specified so in the end it's only rendering the back face, so I thought it was okay. Let me know if I'm being clear or if you need further explanation 🙂